app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -161,14 +143,29 @@ export const assertScopesSupported = (scopes, findNetworkClientIdByChainId) => {
         'Requested notifications are not supported',
       );
     }
+
+    if (accounts) {
+      const accountsSupported = accounts.every((account) =>
+        isSupportedAccount(account, getInternalAccounts),
+      );
+
+      if (!accountsSupported) {
+        // TODO: There is no error code or message specified in the CAIP-25 spec for when accounts are not supported
+        // The below is made up
+        throw new EthereumRpcError(
+          5103,
+          'Requested accounts are not supported',
```

Comment:
this introduces a way for a dapp to check if you have an account without popping up a UI

Reply:
Good point. The only way to not leak this would be to always open the confirmation and only give the reject option to the user. The other way around this is to simply not honor the requested accounts. Which do you think is preferable?

Reply:
I would think that we don't actually want to do validation against the accounts and only honor them (pre-select them in the connection UI) if they're valid. I don't think it needs to block the request. Say a user deletes that accoount in their wallet after having previously used it with a dapp the dapp might request it but then we would be failing the whole connection request when this validation failed.